### DISCUSSION TITLE
[deleted]

### DISCUSSION THREAD

**Discussion by ludoluds** [cypress-io/cypress#22039] [Questions and Help]
[deleted]

**Comment by istateside** [comment]
This is not an ideal solution, you're definitely better off in the long run changing your code to pull from a specific window context rather than `window.top`, but we have a similar situation, and my team solved it by setting the properties we need on the Cypress window when `window.Cypress` is true.

```js
if (window.Cypress) {
  window.top.myImportantFunctions = window.myImportantFunctions;
  window.top.myOtherObject = window.myOtherObject;
}
```

I imagine this puts you at some real risk of memory leaks in your application under test, but it's what we're using to get unblocked while we work on ripping out all of the window.top in our code
